### PR TITLE
Add esp charge and resp charge

### PR DESCRIPTION
**Pull request by wxj6000**

### CODE REVIEW

File: gpu4pyscf/pop/esp.py [hunk 1]
```diff
@@ -0,0 +1,222 @@
...
+    1.36, # Mg
+    1.24, # Al,
+    1.17, # Si,
+    1.80, # P,
+    1.75, # S,
+    1.70]) # Cl
+
+def unit_surface(n):
+    '''
+    Generate spherical harmonics grid points on unit sphere
```
Comment (sunqm) [inline gpu4pyscf/pop/esp.py:55]: 1. It's not spherical harmonics grids. Just one kind of "uniform" grids. Is it right? 
Is there a link or a reference for the grids generation scheme.

2. Why not using Lebedev grids?

File: gpu4pyscf/pop/esp.py [hunk 2]
```diff
@@ -0,0 +1,222 @@
...
+
+        ux.append(x)
+        uy.append(y)
+        uz.append(z*np.ones_like(x))
+    
+    ux = np.concatenate(ux)
+    uy = np.concatenate(uy)
+    uz = np.concatenate(uz)
+
+    return np.array([ux[:n], uy[:n], uz[:n]]).T
```
Comment (sunqm) [inline gpu4pyscf/pop/esp.py:84]: This grid scheme does not necessary produce n grids, right? For example, if n = 30, the output would produce only 23 grids.

File: gpu4pyscf/pop/esp.py (hunk 1 above)
Comment (wxj6000) [reply gpu4pyscf/pop/esp.py:55]: 1. Right. The reference is on the top of the file.
2. For reproducibility. The 'uniform' was used qchem, psi4, and GAMESS. Q-Chem also provides the support of Lebedev grids. We can add the support if requested.

File: gpu4pyscf/pop/esp.py (hunk 2 above)
Comment (wxj6000) [reply gpu4pyscf/pop/esp.py:84]: Right. It generally produces less than n points. Let me add some comments here.

**Pull request merged**